src/render-delaying-services.js

```diff
@@ -56,11 +76,25 @@ const LOAD_TIMEOUT = 3000;
  *     as the detected render delaying services
  */
 export function waitForServices(win) {
-  const promises = includedServices(win).map(service => {
+  const promises = includedServices(win).map(serviceId => {
+
+    const serviceReadyPromise = getServicePromise(
+        win,
+        serviceId
+    ).then(service => {
+      if (service && service.whenReady) {
+        return service.whenReady().then(() => {
+          return service;
+        });
+      }
+      return Promise.resolve(service);
+    });
+
     return Services.timerFor(win).timeoutPromise(
         LOAD_TIMEOUT,
-        getServicePromise(win, service),
-        `Render timeout waiting for service ${service} to be ready.`
+        serviceReadyPromise,
+        `Render timeout waiting for service ${serviceId} ` +
+        'to be ready.'
```

Comment:
Can you remind me if this will cause an exception on timeout?

Reply:
Use doesn't include the extension, or forgets to use an element that defines the service for the extension.

Or, network.

Reply:
It does throw an exception:

![screen shot 2019-03-05 at 2 50 32 pm](https://user-images.githubusercontent.com/1448289/53843060-1e52f000-3f56-11e9-8f73-d3004a7bc72c.png)